src/core/public/application/utils.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Utility to remove trailing, leading or duplicate slashes.
+ * By default will only remove duplicates.
+ */
+export const removeSlashes = (
+  url: string,
+  {
+    trailing = false,
+    leading = false,
+    duplicates = true,
+  }: { trailing?: boolean; leading?: boolean; duplicates?: boolean } = {}
+): string => {
+  if (duplicates) {
+    url = url.replace(/\/{2,}/g, '/');
+  }
+  if (trailing) {
+    url = url.replace(/\/$/, '');
+  }
+  if (leading) {
+    url = url.replace(/^\//, '');
+  }
+  return url;
+};
+
+export const appendAppPath = (appBasePath: string, path: string = '') => {
```

Comment:
are there any tests for `appendAppPath`?

Reply:
My bad, this was introduced in my last commit, and I did not add proper coverage for it. Will add.